kev/backends/dynamodb/db.py

```diff
@@ -0,0 +1,214 @@
+        self.log.debug("getting item: {}".format(doc_id))
+        doc = self._indexer.get_item(Key=doc_id)
+        return doc_class(**(doc["Item"]))
+
+
+    ############################################################################
+    # Desc: empty table
+    #
+    ############################################################################
+    def flush_db(self):
+        # FUTURE: potentially improve efficiency for large tables, delete and recreate table
+
+        # scan table to get item list
+        resp  = self._indexer.scan()
+        items = resp["Items"]
+        self.log.debug("flush_db: found {} items to delete".format(len(items)))
+
+        # delete items using batch delete
+        if len(items) == 0:
+            return
+        with self._indexer.batch_writer() as batch:
+            for item in items:
+                pk = {k: item[k] for k in self._getPK()}
+                self.log.debug("\tflush_db: deleting: {}".format(pk))
+                batch.delete_item(Key=pk)
+
+
+    ############################################################################
+    # Desc: delete single object from table
+    #
+    # @param  doc_obj; document object to be deleted
+    ############################################################################
+    def delete(self, doc_obj):
+        # # TODO why do I need to use save before the delete?
+        # doc_obj2, doc = self._save(doc_obj)
+
+        # create key/value dict of primary key for item to be deleted
+        pk = {k: getattr(doc_obj,k) for k in self._getPK()}
+
+        # delete item
+        self.log.debug("delete: deleting item: {}".format(pk))
+        self._indexer.delete_item(Key=pk)
+
+
+    ############################################################################
+    # Desc: return all currently stored items
+    #
+    # @param  doc_class; document class
+    #
+    # @ret Document DynamoDB object list
+    ############################################################################
+    def all(self,doc_class):
+        # TODO: verify scans > 1MB still work
+        resp = self._indexer.scan()
+        item_list = resp["Items"]
+        self.log.debug("scanning for all items, '{}' found".format(len(item_list)))
+
+        for item in item_list:
+            yield doc_class(**item)
+
+
+    ##############################################################################################
+    # Indexing Methods
+    ##############################################################################################
+
+    ############################################################################
+    # Desc: return items matching provided filter
+    #
+    # @param  filters_list; list of filters
+    # @param  doc_class; document class
+    #
+    # @ret Document DynamoDB object list
+    ############################################################################
+    def evaluate(self, filters_list, doc_class):
```

Comment:
Not sure about this comment. Should I implement a parse_filter function as part of my change?